subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/NormalizedPathFingerprintCompareStrategy.java

```diff
@@ -56,61 +55,138 @@ private NormalizedPathFingerprintCompareStrategy() {
-            FileSystemLocationFingerprint previousFingerprint = unaccountedForPreviousFingerprintEntry.getKey();
-            String normalizedPath = previousFingerprint.getNormalizedPath();
-            List<FilePathWithType> addedFilesForNormalizedPath = addedFilesByNormalizedPath.get(normalizedPath);
-            if (!addedFilesForNormalizedPath.isEmpty()) {
-                // There might be multiple files with the same normalized path, here we choose one of them
-                FilePathWithType addedFile = addedFilesForNormalizedPath.remove(0);
-                DefaultFileChange modified = DefaultFileChange.modified(addedFile.getAbsolutePath(), propertyTitle, previousFingerprint.getType(), addedFile.getFileType(), normalizedPath);
-                if (!visitor.visitChange(modified)) {
-                    return false;
-                }
-            } else {
-                FilePathWithType removedFile = unaccountedForPreviousFingerprintEntry.getValue();
-                DefaultFileChange removed = DefaultFileChange.removed(removedFile.getAbsolutePath(), propertyTitle, removedFile.getFileType(), normalizedPath);
-                if (!visitor.visitChange(removed)) {
-                    return false;
-                }
-            }
-        }
+        return results;
+    }
 
-        if (includeAdded) {
-            for (Map.Entry<String, FilePathWithType> entry : addedFilesByNormalizedPath.entries()) {
-                FilePathWithType addedFile = entry.getValue();
-                DefaultFileChange added = DefaultFileChange.added(addedFile.getAbsolutePath(), propertyTitle, addedFile.getFileType(), entry.getKey());
-                if (!visitor.visitChange(added)) {
-                    return false;
-                }
-            }
-        }
-        return true;
+    private static Change modified(
+        String propertyTitle,
+        FileType previousFingerprintType,
+        String normalizedPath,
+        FilePathWithType modifiedFile
+    ) {
+        String absolutePath = modifiedFile.getAbsolutePath();
+        FileType fileType = modifiedFile.getFileType();
+        return DefaultFileChange.modified(absolutePath, propertyTitle, previousFingerprintType, fileType, normalizedPath);
+    }
+
+    private static Change removed(
+        String propertyTitle,
+        String normalizedPath,
+        FilePathWithType removedFile
+    ) {
+        String absolutePath = removedFile.getAbsolutePath();
+        FileType fileType = removedFile.getFileType();
+        return DefaultFileChange.removed(absolutePath, propertyTitle, fileType, normalizedPath);
+    }
+
+    private static Change added(
+        String propertyTitle,
+        Entry<String, FilePathWithType> addedFilesByNormalizedPathEntries
+    ) {
+        FilePathWithType addedFile = addedFilesByNormalizedPathEntries.getValue();
```

Comment:
Instead of passing the `Entry` pass only the value as parameter.

Reply:
pass the key and value as parameters instead, you mean?